File: frontend/src/types/product.ts
export interface Product {
  id: string | number;
  name: string;
  description: string;
  price: number;
  brand: string;
  sellerId: string | number;
  category: string;
  stock: number;
  carbonFootprint: number;
  sustainablePackaging: boolean;
  rating: number;
  reviews: number;
  imageUrl: string | null;
  isActive: boolean;
  unitsSold: number;
  image?: string; // for backward compatibility
  createdAt?: string;
  updatedAt?: string;
}

export type SortOption = 
  | 'carbon-low'
  | 'carbon-high'
  | 'price-low'
  | 'price-high'
  | 'name';

export interface FilterOptions {
  sortBy: SortOption;
  category?: string;
  maxPrice?: number;
  maxCarbon?: number;
  searchQuery?: string;
  co2Max?: number;
  minCO2?: number;
}

File: backend/src/main/java/com/example/demo/Model/Product.java
package com.example.demo.Model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private Double price;

    private String brand;

    @Column(name = "seller_id")
    private Long sellerId;

    private String category;

    private Integer stock;

    private Float carbonFootprint;

    private Boolean sustainablePackaging;

    private Float rating;

    private String imageUrl;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private Boolean isActive;

    private Integer unitsSold;


    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (isActive == null) isActive = true;
        if (sustainablePackaging == null) sustainablePackaging = false;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
